scripts/speech_recognition/convert_to_tarred_audio_dataset.py

```diff
@@ -686,14 +670,56 @@ def _create_shard(self, entries, target_dir, shard_id, manifest_folder):
             # Need the following replacement as long as WebDataset splits on first period
             base = base.replace('.', '_')
             squashed_filename = f'{base}{ext}'
-            if squashed_filename not in count:
-                self._write_to_tar(tar, audio_filepath, squashed_filename)
-                to_write = squashed_filename
-                count[squashed_filename] = 1
-            else:
-                to_write = base + "-sub" + str(count[squashed_filename]) + ext
+
+            if self.config.slice_with_offset:
+                if squashed_filename not in count:
+                    count[squashed_filename] = 1
+
+                entry_offset = str(entry['offset']).split('.')
+                if len(entry_offset) == 1:
+                    # Example: offset = 12 -> becomes 12_0
+                    entry_offset.append('0')
+                elif len(entry_offset) == 2:
+                    # Example: offset = 12.34 -> becomes 12_34
+                    pass
+                else:
+                    raise ValueError(
+                        f"The offset for the entry with audio_filepath '{entry['audio_filepath']}' is incorrectly provided ({entry['offset']}). "
+                        "Expected a float-like value (e.g., 12 or 12.34)."
+                    )
+                entry_offset = "_".join(entry_offset)
+
+                entry_duration = str(entry['duration']).split('.')
+                if len(entry_duration) == 1:
+                    entry_duration.append('0')
+                elif len(entry_duration) > 2:
+                    raise ValueError(
+                        f"The duration for the entry with audio_filepath '{entry['audio_filepath']}' is incorrectly provided ({entry['duration']})."
+                    )
+                entry_duration = "_".join(entry_duration)
+
+                to_write = base + "_" + entry_offset + "_" + entry_duration + ext
+                if not only_manifests:
+                    self._write_to_tar(
+                        tar, audio_filepath, to_write, duration=entry['duration'], offset=entry['offset']
+                    )
                 count[squashed_filename] += 1
 
+                entry['source_audio_offset'] = entry['offset']
+                del entry['offset']
+            else:
+                if squashed_filename not in count:
+                    if not only_manifests:
+                        self._write_to_tar(tar, audio_filepath, squashed_filename)
+                    to_write = squashed_filename
+                    count[squashed_filename] = 1
+                else:
+                    to_write = base + "-sub" + str(count[squashed_filename]) + ext
+                    count[squashed_filename] += 1
```

Comment:
Do we need this now? Previously we used -sub for writing offset audio files, and now you are giving the script an explicit option which was handled in above case.

Reply:
The reason for sub is, in nemo tarred datasets, I believe we use this to check if we need to segment the file with  an offset or not. I would recommend you to add a test with a tarred set created with offset with updated set and make sure dataloading is working as intedted with both lhotse and nemo tarred dataset